alacritty/src/window.rs

```diff
@@ -254,22 +262,26 @@ impl Window {
 
         let class = &window_config.class;
 
-        let mut builder = WindowBuilder::new()
+        let builder = WindowBuilder::new()
             .with_title(title)
             .with_visible(false)
             .with_transparent(true)
             .with_decorations(window_config.decorations != Decorations::None)
             .with_maximized(window_config.maximized())
             .with_fullscreen(window_config.fullscreen())
-            .with_window_icon(icon.ok())
-            // X11.
-            .with_class(class.instance.clone(), class.general.clone())
-            // Wayland.
-            .with_app_id(class.instance.clone());
+            .with_window_icon(icon.ok());
 
-        if let Some(ref val) = window_config.gtk_theme_variant {
-            builder = builder.with_gtk_theme_variant(val.clone())
-        }
+        // Wayland.
+        #[cfg(feature = "wayland")]
+        let builder = builder.with_app_id(class.instance.clone());
+        #[cfg(feature = "x11")]
+        let builder = builder.with_class(class.instance.clone(), class.general.clone());
+
+        #[cfg(feature = "x11")]
+        let builder = match &window_config.gtk_theme_variant {
+            Some(val) => builder.with_gtk_theme_variant(val.clone()),
+            None => builder,
+        };
```

Comment:
I think the `if let Some` used previously is a bit nicer here?

Reply:
We can't use `mut`, since we can't put cfg above `builder = builder`, so `match` is shorter than `if let`.